extension-manifest-v3/src/background/stats.js

```diff
@@ -99,21 +99,18 @@ export async function updateTabStats(tabId, requests) {
   // as some of the requests are fired before the tab is created, tabId -1
   if (!stats) return;
 
-  const filtered = requests.filter(
-    ({ url }) => !stats.trackers.some((t) => t.url === url),
-  );
+  for (const request of requests) {
+    const pattern = await trackerDb.getMetadata(request);
```

Comment:
pass additional argument to match patterns by their domain - this will improve detection https://github.com/ghostery/adblocker/blob/7f652911ad40b2572f10144091bc7e413caee270/packages/adblocker/src/engine/engine.ts#L932